### PR TITLE
IODA Version 3

### PR DESCRIPTION
**Pull request by danholdaway**
## Description

Update all the tests to IODA V3. Updates test files and YAMLs and source code where necessary. Where I found reliance on hardcoded names for channel or location outside of the IODA obs space class I tried to make it more generic.

Note that now the name for Location differs between the IODA and GSI classes so a new method is needed to change it to Location. I figure the default should be what it in IODA. The GSI class now has to call the method. In order to use the `sel` function in IODA you need to know the variable name ahead of time, hence why the Data class has to have a specific name.

### CODE REVIEW

File: src/eva/data/data_collections.py [hunk 1]
```diff
@@ -230,6 +241,7 @@ def display_collections(self):
         self.logger.info('-'*80)
         self.logger.info(fcol.bold + 'Collections available: ' + fcol.end)
         for collection in self._collections.keys():
+            print(self._collections[collection])
```
Comment (CoryMartin-NOAA) [inline src/eva/data/data_collections.py]: is this for debugging or intended to stay?

**Review by CoryMartin-NOAA — COMMENTED** [review]
minor question/comment

File: src/eva/data/data_collections.py (hunk 1 above)
Comment (danholdaway) [reply src/eva/data/data_collections.py]: Good catch. Thanks.

Comment (danholdaway) [inline src/eva/data/data_collections.py]:
```suggestion
```

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**